building fails on mac

cant really test myself, but github actions gave me this:

```
  Error: /Users/runner/work/abaddon/abaddon/subprojects/keychain/src/keychain_mac.cpp:115:23: error: 'SecKeychainFindGenericPassword' is deprecated: first deprecated in macOS 10.10 - SecKeychain is deprecated [-Werror,-Wdeprecated-declarations]
      OSStatus status = SecKeychainFindGenericPassword(
                        ^
  /Applications/Xcode_14.2.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX13.1.sdk/System/Library/Frameworks/Security.framework/Headers/SecKeychain.h:628:10: note: 'SecKeychainFindGenericPassword' has been explicitly marked deprecated here
  OSStatus SecKeychainFindGenericPassword(CFTypeRef __nullable keychainOrArray,  UInt32 serviceNameLength, const char * __nullable serviceName, UInt32 accountNameLength, const char * __nullable accountName, UInt32 * __nullable passwordLength, void * __nullable * __nullable passwordData, SecKeychainItemRef * __nullable CF_RETURNS_RETAINED itemRef)
           ^
```

maybe a -Wno-error=deprecated-declarations would suffice?
https://github.com/hrantzsch/keychain/blob/f0e4d49e561f4639efd33f0798840f90dc2b2c18/CMakeLists.txt#L25

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -45,6 +45,12 @@ elseif (APPLE)
         PRIVATE
             "src/keychain_mac.cpp")
 
+    # We're using the deprecated "Legacy Password Storage" API
+    # (SecKeychainFindGenericPassword and friends)
+    target_compile_options(${PROJECT_NAME}
+        PRIVATE
+            "-Wno-error=deprecated-declarations")
+
     find_library(COREFOUNDATION_LIBRARY CoreFoundation REQUIRED)
     find_library(SECURITY_LIBRARY Security REQUIRED)
 

